doc/source/scripts/gen_topologyparser_attrs.py

```diff
@@ -64,101 +65,170 @@
 )
 
 
-MANDATORY_ATTRS = set(mandatory_attrs)
-
-
-parser_attrs = {}
-
-
-for p in PARSER_TESTS:
-    e, g = set(p.expected_attrs) - MANDATORY_ATTRS, set(p.guessed_attrs)
-    # clunky hack for PDB
-    if p is TestPDBParser:
-        e.add("elements")
-    parser_attrs[p.parser] = (e, g)
-
-
-class TopologyParsers(TableWriter):
-    headings = ["Format", "Description", "Attributes read", "Attributes guessed"]
-    preprocess = ["keys"]
-    filename = "formats/topology_parsers.txt"
-    include_table = "Table of supported topology parsers and the attributes read"
-    sort = True
-
-    def __init__(self):
-        self.attrs = defaultdict(set)
-        super(TopologyParsers, self).__init__()
-
-    def _set_up_input(self):
-        return [[x, *y] for x, y in parser_attrs.items()]
-
-    def get_line(self, parser, expected, guessed):
-        line = super(TopologyParsers, self).get_line(parser, expected, guessed)
-        for a in expected | guessed:
-            self.attrs[a].add(self.fields["Format"][-1])
-        return line
-
-    def _keys(self, parser, *args):
-        f = parser.format
-        if isinstance(f, (list, tuple)):
-            key = f[0]
-            label = ", ".join(f)
-        else:
-            key = label = f
-        return (key, label)
-
-    def _description(self, *args):
-        key, label = self.keys[-1]
-        return DESCRIPTIONS[key]
+def create_parser_attributes() -> dict[Any, tuple[set[str], set[str]]]:
+    parser_attrs = {}
+    for test_parser_class in PARSER_TESTS:
+        expected, guessed = set(test_parser_class.expected_attrs) - set(
+            mandatory_attrs
+        ), set(test_parser_class.guessed_attrs)
```

Comment:
```suggestion
        expected = set(test_parser_class.expected_attrs) - set(mandatory_attrs)
        guessed = set(test_parser_class.guessed_attrs)
```

IMO this is easier to read than trying to have them in the same definition.